src/tests/api/CallbackApi.test.ts

```diff
@@ -328,7 +328,59 @@ describe("Callback API", () => {
 		do {
 			sqsMessage = await getSqsEventList("txma/", sessionId, 7);
 		} while (!sqsMessage);
-		await validateTxMAEvent("F2F_CRI_VC_ISSUED", sqsMessage, yotiMockId);
+		await validateTxMAEvent("F2F_CRI_VC_ISSUED", sqsMessage, yotiMockId, false, vcResponseData);
+
+	}, 20000);
+
+	it.each([
+		["0121", dataPassport],
+		["0109", dataPassport],
+		["0110", dataPassport],
+		["0111", dataPassport],
+		["0112", dataPassport],
+		["0113", dataPassport],
+		["0114", dataPassport],
+		["0115", dataPassport],
+		["0118", dataPassport],
+		["0119", dataPassport],
+		["0120", dataPassport],
+		["0122", dataPassport],
+		["0204", dataNonUkPassport],
+
+	])("F2F CRI F2F_CRI_VC_ISSUED TxMA Validation - Contra Indicators - yotiMockId: '%s'", async (yotiMockId: string, docSelectionData:any) => {
+		f2fStubPayload.yotiMockID = yotiMockId;
+		const sessionResponse = await startStubServiceAndReturnSessionId(f2fStubPayload);
+		const sessionId = sessionResponse.data.session_id;
+		const sub = sessionResponse.data.sub;
+
+		// Document Selection
+		const response = await postDocumentSelection(docSelectionData, sessionId);
+		expect(response.status).toBe(200);
+		// Authorization
+		const authResponse = await authorizationGet(sessionId);
+		expect(authResponse.status).toBe(200);
+		// // Post Token
+		const tokenResponse = await tokenPost(authResponse.data.authorizationCode.value, authResponse.data.redirect_uri );
+		expect(tokenResponse.status).toBe(200);
+		// Post User Info
+		const userInfoResponse = await userInfoPost("Bearer " + tokenResponse.data.access_token);
+		expect(userInfoResponse.status).toBe(202);
+
+		// Get Yoti Session Id
+		const session = await getSessionById(sessionId, constants.DEV_F2F_SESSION_TABLE_NAME);
+		const yotiSessionId: any = session?.yotiSessionId;
+		console.log(yotiSessionId);
```

Comment:
A suggestion (and feel free to ignore if you like) but these initial steps are repeated quite a lot, is it work putting them into a beforeEach?

Reply:
I had a look at this and don't think any of the logic can be put into a beforeEach step. 

The repeated logic to start the stub and return the sessionId can't be put into a beforeEach as we'll need the yotiMockId variable to update the shared_claims payload before we start the stub and this variable won't be available to us before the test starts running. 

With get Yoti Session Id, we can only kick this off after the call to the /documentSelection endpoint when we have generated a yotiMockId for this session so can't be put in a beforeEach as that yotiMockId value will not exist before a test is ran

Reply:
Sorry not sure I understand - could you not assign `yotiMockId` in the beforeEach too like [this](https://github.com/govuk-one-login/ipv-cri-f2f-api/blob/KIWI-1267-sigv4/src/tests/api/DocumentSelectionHappyPath.test.ts#L13)?